da/celestia/celestia.go

```diff
@@ -172,3 +174,19 @@ func (c *DataAvailabilityLayerClient) RetrieveBlocks(ctx context.Context, dataLa
 		Blocks: blocks,
 	}
 }
+
+func dataRequestErrorToStatus(err error) da.StatusCode {
```

Comment:
this needs a unit test

Reply:
@MSevey it is not possible to test different status messages using our mock server because it does not differentiate namespaces while creating the mock DALC: https://github.com/rollkit/rollkit/blob/main/da/celestia/mock/server.go#L47
This means, all namespace data gets stored into a single key-value store and querying for different namespaces would result in the same data.
Extending the mock server to create one mock DALC for each namespace would be too large scope for this issue. Hence, created a new issue to address that part: https://github.com/rollkit/rollkit/issues/993

Reply:
I don't think you need to extend anything.  I'm talking basic unit test functionality of this error switch handling.

We can write a simple table test the verifies given a specific error, we get the expect response code. No mocks needed as we are dealing with just variables.